src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Finder\Iterator;
+
+/**
+ * Reverse the order of a previous iterator.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class ReverseSortingIterator extends \ArrayIterator
+{
+    public function __construct(\Traversable $iterator)
+    {
+        parent::__construct(array_reverse(iterator_to_array($iterator, true)));
```

Comment:
Instead of doing the reversing in the constructor, it should be done in a `IteratorIterator::getIterator()` method, like the `SortableIterator` does.
Which makes me wonder: should this concern be added to `SortableIterator` instead? Looks very related to me.

Reply:
You are right about the constructor vs `getIterator`. I have fixed the code.

About moving it to SortableIterator, I don't think it's a good idea
1. If you don't sort, you can not reverse the sort. This is a bit weird, but if people want to reverse the "natural sorting" with this implemention, they could.
2. Composition is a better design